graph_generator/node.py

```diff
@@ -238,3 +241,74 @@ def __gt__(self, other):
 
     def __str__(self):
         return f"Node: {self.config.name}"
+
+
+"""
+Load YAML configuration files and construct the appropriate configuration 
+objects for nodes, callbacks, and publishing in a messaging system.
+"""
+
+
+class LoadConfigError(Exception):
+    pass
+
+
+def convert_to_tuple(data: List[int]) -> Tuple[int, int]:
+    if not isinstance(data, list) or len(data) != 2:
+        raise LoadConfigError(f"Expected a list of two integers, got: {data}")
+    return tuple(data)
+
+
+def construct_publish_config(data: dict) -> PublishConfig:
+    data["value_range"] = convert_to_tuple(data["value_range"])
+    data["delay_range"] = convert_to_tuple(data["delay_range"])
+    return PublishConfig(**data)
+
+
+def construct_callback_config(data: dict) -> CallbackConfig:
```

Comment:
Is there anyway to parse yaml in a generic way? For example, if we add more config fields or new config types, is it possible to get away with not changing the parser?